src/main/java/com/snowflake/kafka/connector/internal/streaming/telemetry/SnowflakeTelemetryChannelStatus.java

```diff
@@ -18,33 +18,45 @@
 package com.snowflake.kafka.connector.internal.streaming.telemetry;
 
 import static com.snowflake.kafka.connector.internal.metrics.MetricsUtil.constructMetricName;
+import static com.snowflake.kafka.connector.internal.streaming.TopicPartitionChannel.NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE;
 
 import com.codahale.metrics.Gauge;
 import com.codahale.metrics.MetricRegistry;
-import com.snowflake.kafka.connector.internal.KCLogger;
+import com.google.common.annotations.VisibleForTesting;
 import com.snowflake.kafka.connector.internal.metrics.MetricsJmxReporter;
 import com.snowflake.kafka.connector.internal.metrics.MetricsUtil;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryBasicInfo;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryService;
+import com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants;
 import java.util.concurrent.atomic.AtomicLong;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode;
 
-public class SnowflakeTelemetryChannelStatus {
-  private static final KCLogger LOGGER =
-      new KCLogger(SnowflakeTelemetryChannelStatus.class.toString());
-
-  public static final long NUM_METRICS = 3; // update when new metrics are added
+/**
+ * Extension of {@link SnowflakeTelemetryBasicInfo} class used to send data to snowflake when the
+ * TopicPartitionChannel closes. Also creates and registers various metrics with JMX
+ *
+ * <p>Most of the data sent to Snowflake is an aggregated data.
+ */
+public class SnowflakeTelemetryChannelStatus extends SnowflakeTelemetryBasicInfo {
+  public static final long NUM_METRICS = 4; // update when new metrics are added
 
   // channel properties
   private final String channelName;
   private final MetricsJmxReporter metricsJmxReporter;
+  private final long startTime;
 
   // offsets
   private final AtomicLong offsetPersistedInSnowflake;
   private final AtomicLong processedOffset;
   private final AtomicLong latestConsumerOffset;
 
+  // channel metrics
+  private final AtomicLong channelTryOpenCount;
```

Comment:
Umm, should we rename it something else? this is more of a channelReopenCount in current partition usage..
This will be super confusing with our channel sequencers.. at any given point in time channelSequencers >= channelTryOpenCount

This value will be reset to 0 when there is a rebalance right?
I do think this will be useful in cases during rebalance..
May be rename this to `channelReopenCountInActiveOpenTask` 

I cant think of a better name though, open to suggestions.

Reply:
removed variable, n/a